torch_xla/experimental/xla_sharding.py

```diff
@@ -461,7 +461,8 @@ def wrap_if_sharded(x: Any) -> Any:
   Otherwise, returns the input.
   """
   if (isinstance(x, torch.Tensor) and not isinstance(x, XLAShardedTensor) and
-      torch_xla._XLAC._get_xla_sharding_type(x) is not None):
+      torch_xla._XLAC._get_xla_sharding_type(x) is not None and
+      x.device.type == 'xla'):
```

Comment:
We need to do the `device.type` check first, `_get_xla_sharding_type` asserts that the device is an XLA device